docs/dictionary/property/executionContexts.lcdoc

```diff
@@ -38,7 +38,8 @@ A common use of the <executionContexts> is to obtain the name of the
 object and handler that called the current handler, this information is
 available as: line -2 of the executionContexts.
```

Comment:
@mwieder when there is a behavior object this line is incorrect. I'm wondering if given the complexity of parsing a line of execution contexts in a way that handles potential commas in file names if we should provide a context to array function here.

Reply:
Thanks. I have the other things taken care of now (wasn't sure about the markdown effects), but for this one: you are correct in that this should be line -1, and I have a function that properly retrieves the correct information, as I needed to do this for PowerDebug.

Commas in filenames are not IMO a thing to be handled in a doc file. Personally I'd love it if commas *inside quoted strings* would not be parsed as item separators, but I think that's a separate topic. I like where the context-to-array thing is headed, but it needs further thought to preserve the ordering of lines in the executionContexts.

Reply:
Sorry @mwieder I had a brain fart here... line -2 is correct for the caller context. For some reason I was reading item here... The reason I mention commas in filenames is we have a number of things that return lists with a long id as an item and they always need careful parsing. One day I would like to replace all of them with array list variants and deprecate anything returning a multi-item list with a long id as an item but that won't happen for a while. Hence documenting safe parsing of execution contexts seemed like a good idea.

Reply:
Hmmm... I'm using line -1 for both normal and behavior contexts. Ah! inside the errorDialog it would be -2... I'll correct the function.

Yes... array list variants (keeping the proper order) would be great. Any suggestions on parsing if there are embedded commas? It's a sticky situation.

Reply:
-1 is the current context -2 is the caller.

you need to do something like:

```
local tKey = "object", tContextA
repeat for each item tItem in tContext
    if tContextA[tKey] is not empty then
        put comma after tContextA[tKey]
    end if
    put tItem after tContextA[tKey]
    
    switch tKey
       case "object"
           if exists(tContextA[tKey]) then
                put "handler" into tKey
           end if
           break
       case "handler"
           put "line" into tKey
           break
        case "line"
           put "behavior" into tKey
           break
    end switch
end repeat
```

Reply:
Hey - that's pretty clever. I'd still rather have the comma-inside-quotes thing fixed, but that will do nicely for now.